src/layers/core/grid-layer/grid-layer-vertex.glsl

```diff
@@ -43,41 +29,42 @@ varying vec4 vColor;
 
 void main(void) {
 
-
   // cube gemoetry vertics are between -1 to 1, scale and transform it to between 0, 1
-  vec2 ptPosition = instancePositions.xy + vec2((positions.x + 1.0 ) * lonOffset / 2., (positions.y + 1.) * latOffset / 2.);
+  vec2 ptPosition = instancePositions.xy + vec2((positions.x + 1.0 ) * lonOffset / 2.0, (positions.y + 1.0) * latOffset / 2.0);
 
   vec2 pos = project_position(ptPosition);
 
-  // if 3d not enabled set elevation to 0
-  float elevation =  mix(0., project_scale(instancePositions.w  * (positions.z + 1.) * ELEVATION_SCALE), extruded);
```

Comment:
Would be good to get a solid conclusion on `if` vs `mix` (solid = perf testing on a couple of platforms) as it could help us establish guidelines for how we write shaders in general. So far we have preferred `mix` over `if` but we don't have any data to back it up.